core/modules/filter/filter.pages.inc

```diff
@@ -100,6 +100,25 @@ function _filter_link_autocomplete($string) {
       $matches['/' . $path] = check_plain($result->title) . ' (' . t('Content') . ')';
       $match_count++;
     }
+    
+    				// Get file matches.
+    $results = db_select('file_managed')
+      ->fields('file_managed', array('fid', 'filename', 'uri', 'filemime', 'status'))
+      ->condition('filename', db_like($string) . '%', 'LIKE')
+      ->condition('status', 1)
+						->condition('filemime', 'image%', 'NOT LIKE')
+      ->orderBy('fid', 'DESC')
+      ->range(0, 10)
```

Comment:
`range` should handle `$max_count` as for users, see https://github.com/backdrop/backdrop/pull/2084/files#diff-8d9f5bbb77a85ec42c32392f64402cc4R128

Reply:
Thanks @opi I hadn't spotted this.